rimage/transform/parallel_projection_test.go

```diff
@@ -0,0 +1,208 @@
+
+		pointcloud := pc.New()
+		err := pointcloud.Set(r3.Vector{X: 0, Y: 0, Z: 0}, pc.NewValueData(200))
+		test.That(t, err, test.ShouldBeNil)
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(pointcloud)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring,
+			fmt.Sprintf("received a value of %v which is outside the range (0 - 100) representing probabilities", 200))
+		test.That(t, im, test.ShouldBeNil)
+		test.That(t, unusedDepthMap, test.ShouldBeNil)
+	})
+
+	t.Run("Project a two point pointcloud with data with image pixel checks", func(t *testing.T) {
+		pose := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
+		ppRM := NewParallelProjectionOntoXZWithRobotMarker(&pose)
+
+		pointcloud := pc.New()
+		d := pc.NewBasicData()
+		p1 := r3.Vector{X: -2, Y: 10, Z: -3}
+		err := pointcloud.Set(p1, d)
+		test.That(t, err, test.ShouldBeNil)
+		p2 := r3.Vector{X: 10, Y: 10, Z: 10}
+		err = pointcloud.Set(p2, d)
+		test.That(t, err, test.ShouldBeNil)
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(pointcloud)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, im.Width(), test.ShouldEqual, imageWidth)
+		test.That(t, im.Height(), test.ShouldEqual, imageHeight)
+		test.That(t, unusedDepthMap, test.ShouldBeNil)
+
+		minX := math.Min(math.Min(pose.Point().X, p1.X), p2.X)
+		maxX := math.Max(math.Max(pose.Point().X, p1.X), p2.X)
+		minZ := math.Min(math.Min(pose.Point().Z, p1.Z), p2.Z)
+		maxZ := math.Max(math.Max(pose.Point().Z, p1.Z), p2.Z)
+
+		scaleFactor := (math.Min((imageWidth-1)/(maxX-minX), (imageHeight-1)/(maxZ-minZ)))
+
+		robotMarkerExpectedPos := image.Point{
+			X: int(math.Round((pose.Point().X - minX) * scaleFactor)),
+			Y: int(math.Round((pose.Point().Z - minZ) * scaleFactor)),
+		}
+
+		colorAtPos := im.GetXY(robotMarkerExpectedPos.X, robotMarkerExpectedPos.Y)
+		expectedRobotMarkerColor := rimage.NewColor(255, 0, 0)
+		test.That(t, colorAtPos, test.ShouldResemble, expectedRobotMarkerColor)
+
+		point1ExpectedPos := image.Point{
+			X: int(math.Round((p1.X - minX) / scaleFactor)),
+			Y: int(math.Round((p1.Z - minZ) / scaleFactor)),
+		}
+
+		colorAtPoint1 := im.GetXY(point1ExpectedPos.X, point1ExpectedPos.Y)
+		expectedPoint1Color, err := getProbabilityColorFromValue(d)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, colorAtPoint1, test.ShouldResemble, expectedPoint1Color)
+
+		point2ExpectedPos := image.Point{
+			X: int(math.Round((p2.X - minX) / scaleFactor)),
+			Y: int(math.Round((p2.Z - minZ) / scaleFactor)),
+		}
+
+		colorAtPoint2 := im.GetXY(point2ExpectedPos.X, point2ExpectedPos.Y)
+		expectedPoint2Color, err := getProbabilityColorFromValue(d)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, colorAtPoint2, test.ShouldResemble, expectedPoint2Color)
+	})
+
+	t.Run("Project an imported pointcloud", func(t *testing.T) {
+		p := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
```

Comment:
Is is possible to write a test that takes all of `"pointcloud/test.pcd"` & compares the returned `image` to the expected image (which could also be stored in artifacts)?

Also could we get a test that involves point clouds with both positive & negative values & confirm that the resulting image is the same as if the same points had the same relation to each other, but were all positive?  

e.g. `[(-10,-5), (0,0), (10,5)]` should render the same image as `[(0,0), (10,5), (20,15)]` right?

Reply:
Adding the second test. First one fails due to encoding decode altering the images to the extend the ShouldResemble cannot compare. We could compare several points of interest but i think the second one is sufficient

Reply:
I am adding my 2 cents: I favored both options if they were easily possible. We should continue thinking about how to test visual output like this in the future and consider building some tools.